### ISSUE TITLE
Why are the text and image feature inputs treated as unfrozen embeddings?

### ISSUE THREAD

**Issue by gmeehan96** [enoche/MMRec#56]
Hi - thanks for the really useful framework. I'm confused by the choice in the code for many of the multimodal models to load the image and text features in as unfrozen embeddings, e.g.:

`self.image_embedding = nn.Embedding.from_pretrained(self.v_feat, freeze=False)`

While these are detached from gradients in the construction of the KNN graphs etc., they are treated as trainable variables for all models which have any MLP or linear transformation of the input features, e.g. [MGCN](https://github.com/demonph10/MGCN/blob/c187ee85b68a3275c97ce730361940db11690072/src/models/mgcn.py#L149C13-L150C1): 

`image_feats = self.image_trs(self.image_embedding.weight)`

My understanding is that the text/image features should be treated as fixed input tensors and stay constant during training. However, since the embeddings are not frozen, they will be updated during training - I've verified this by training a PGL model on the Sports dataset and comparing the model weights before and after training is complete. These are the image embedding weights before:

<img width="530" height="141" alt="Image" src="https://github.com/user-attachments/assets/8442afd5-d7d2-4abf-b9f1-368f1ccea320" />

and after training:

<img width="544" height="140" alt="Image" src="https://github.com/user-attachments/assets/fccf3e20-8f2f-44b2-9df4-38f9195041a7" />

This seems like a strange use of the content features, as they're essentially treated like a separate set of learnable item ID embeddings (in some cases, with very large dimensions) and may lose most of the original information about visual or textual content due to gradient updates during training.   As far as I can tell, this choice is also not made clear in any of the relevant papers for these models. Please could you clarify if this is a deliberate design choice?

Thanks!

**Comment by enoche** [comment]
Hi, @gmeehan96 Thank you for your feedback on this repository. I find your concern intriguing and will look into the issue further.
Current code generally follows LATTICE. `https://github.com/enoche/MMRec/blob/master/src/models/lattice.py`

**Comment by SmartZhuoCho** [comment]
@gmeehan96 
Hello, I am a master's student researching multimodal recommendation.
The question you raised is excellent; in fact, I have also grappled with this issue myself. I wonder if you have any unique insights on the matter? I would greatly appreciate your guidance.

**Comment by gmeehan96** [comment]
Hi @SmartZhuoCho - in my personal implementation, I have changed the freeze parameter to True for all affected models for subsequent work. This change generally hasn't had a significant impact on the performance of most of them, so I don't think this potential error means that the results from the papers are necessarily misleading. But I haven't conducted a robust analysis of this.

@enoche - have you had the chance to look into the issue?
